omod-1.12/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderGroupResource1_12.java

```diff
@@ -0,0 +1,214 @@
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+import io.swagger.models.Model;
+import io.swagger.models.ModelImpl;
+import io.swagger.models.properties.ArrayProperty;
+import io.swagger.models.properties.BooleanProperty;
+import io.swagger.models.properties.RefProperty;
+import io.swagger.models.properties.StringProperty;
+
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+
+@Resource(name = RestConstants.VERSION_1 + "/orderGroup", supportedClass = OrderGroup.class, supportedOpenmrsVersions = {
+        "1.12.*", "2.0.*", "2.1.*", "2.2.*" })
+public class OrderGroupResource1_12 extends DataDelegatingCrudResource<OrderGroup> {
+	
+	@Override
+	public OrderGroup newDelegate() {
+		return new OrderGroup();
+	}
+	
+	@Override
+	public OrderGroup save(OrderGroup delegate) {
+		return Context.getOrderService().saveOrderGroup(delegate);
+	}
+	
+	@Override
+	public OrderGroup getByUniqueId(String uniqueId) {
+		
+		return Context.getOrderService().getOrderGroupByUuid(uniqueId);
+	}
+	
+	@Override
+	protected void delete(OrderGroup delegate, String reason, RequestContext context) throws ResponseException {
+		if (delegate.isVoided()) {
+			return;
+		}
+		
+		for (Order order : delegate.getOrders()) {
+			Context.getOrderService().voidOrder(order, reason);
+		}
+	}
+	
+	@Override
+	public void purge(OrderGroup delegate, RequestContext context) throws ResponseException {
+		for (Order order : delegate.getOrders()) {
+			Context.getOrderService().purgeOrder(order);
+		}
+		
+	}
+	
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		
+		if (rep instanceof DefaultRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("display");
+			description.addProperty("uuid");
+			description.addProperty("voided");
+			description.addProperty("patient", Representation.REF);
+			description.addProperty("encounter", Representation.REF);
+			description.addProperty("orders", Representation.REF);
+			description.addProperty("orderSet", Representation.REF);
+			description.addSelfLink();
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+			return description;
+			
+		}
+		else if (rep instanceof FullRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("display");
+			description.addProperty("voided");
+			description.addProperty("auditInfo");
+			description.addProperty("patient", Representation.DEFAULT);
+			description.addProperty("encounter", Representation.DEFAULT);
+			description.addProperty("orders", Representation.DEFAULT);
+			description.addProperty("orderSet", Representation.DEFAULT);
+			description.addSelfLink();
+			return description;
+		}
```

Comment:
I can see some code duplication here. How about creating one method to return the DelegatingResourceDescription instance and pass the Representation type accordingly ?

Reply:
nno @dilanthas , thats how we write it .its like the standard way we do it